barretenberg_wasm/src/schnorr.rs

```diff
@@ -1,49 +1,73 @@
 use std::convert::TryInto;
 use wasmer::Value;
 
-use super::Barretenberg;
+use super::{Barretenberg, FIELD_BYTES, SIG_BYTES};
+
 impl Barretenberg {
     pub fn construct_signature(&mut self, message: &[u8], private_key: [u8; 32]) -> [u8; 64] {
-        self.transfer_to_heap(&private_key, 64);
-        self.transfer_to_heap(message, 96);
-        let message_len = Value::I32(message.len() as i32);
+        let message_ptr: usize = 96;
+        let private_key_ptr: usize = 64;
+        let sig_s_ptr: usize = 0;
+        let sig_e_ptr: usize = 32;
+        let result_ptr: usize = 0;
+
```

Comment:
can we add comments here to say why these numbers are the way they are? 

Also I think this function is slightly confusing if you stare at it long enough: 

The result of construct_signature are two field elements (s,e) , so it makes sense that we have two pointers `sig_s` and `sig_e` -- so then why do we have `result_ptr`. 

It would be a bit clearer if we did: `let result_ptr = sig_s_ptr`. This implicitly relies on the fact that we are storing `sig_s` and `sig_e` right next to each other, s is [0,32) and e is [32,64). 

I would probably just do the following:

```rust
  let sig_s_bytes = self.slice_memory(sig_s_ptr, sig_s_ptr + FIELD_BYTES);
  let sig_e_bytes = self.slice_memory(sig_e_ptr, sig_e_ptr + FIELD_BYTES);
```

Then just return (s, e) and in verify_signature we could pass it (s,e) and then there is no need to split

Reply:
The change in API was only a suggestion though I think we should change `let result_ptr: usize = 0;` because it must always be equal to sig_s_ptr

Reply:
I think for the interests of merging this back into `acvm-backend-barretenberg` I'm going to leave the interface the same as it is currently and we can address this up there.